src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1135,4 +1102,64 @@ InferenceProfiler::AllMPIRanksAreStable(bool current_rank_stability)
   return all_stable;
 }
 
+#ifndef DOCTEST_CONFIG_DISABLE
+class TestInferenceProfiler {
+ public:
+  static void ValidLatencyMeasurement(
+      const std::pair<uint64_t, uint64_t>& valid_range,
+      size_t& valid_sequence_count, size_t& delayed_request_count,
+      std::vector<uint64_t>* latencies, TimestampVector& all_timestamps)
+  {
+    InferenceProfiler inference_profiler{};
+    inference_profiler.all_timestamps_ = all_timestamps;
+    inference_profiler.ValidLatencyMeasurement(
+        valid_range, valid_sequence_count, delayed_request_count, latencies);
+  }
+};
+
+TEST_CASE("testing the ValidLatencyMeasurement function")
+{
+  size_t valid_sequence_count{};
+  size_t delayed_request_count{};
+  std::vector<uint64_t> latencies{};
+
+  const std::pair<uint64_t, uint64_t> window{4, 17};
+  TimestampVector all_timestamps{
+      // request ends before window starts, which means it ended after previous
```

Comment:
is this comment correct? Is the assumption that if it was part of the previous window it would have been deleted before now?

Reply:
Correct.

This assumption is implemented to be true by the fact that we always loop [here](https://github.com/triton-inference-server/client/blob/matthewkotila-dropped-requests/src/c%2B%2B/perf_analyzer/inference_profiler.cc#L811) starting at the beginning of all requests, and then that the only condition for a request to be included in a window is to end before the end of the current window [here](https://github.com/triton-inference-server/client/blob/matthewkotila-dropped-requests/src/c%2B%2B/perf_analyzer/inference_profiler.cc#L818).

This guarantees that on each pass, if there exists a request that ended before the current window started, it means it ended after the previous window ended. And we treat that gap between windows as a necessity of the fact that computation isn't instant, but conceptually part of the next window. Also explained [here](https://github.com/triton-inference-server/client/pull/113#discussion_r890633962).

Reply:
ok, just add a little more info to the comment in the test then.

// request ends before window starts, **but the fact that it exists in the list** means it ended after ...

Reply:
Updated comment 👍